pkg/scanners/scannerv4/convert.go

```diff
@@ -191,50 +198,123 @@ func setScoresAndScoreVersion(vuln *storage.EmbeddedVulnerability, vulnCVSS *v4.
-			vuln.ScoreVersion = storage.EmbeddedVulnerability_V2
-			vuln.Cvss = c.Score
+			vuln.ScoreVersion = sv
+			vuln.Cvss = score
+			vuln.CvssV2 = cvssV2
+		}
+	}
+
+	if vulnCVSS.GetV3() != nil {
+		score, sv, cvssV3, err := toCVSSV3Scores(vulnCVSS, vuln.GetCve())
+		if err != nil {
+			errList.AddError(err)
 		} else {
-			errList.AddError(fmt.Errorf("v2: %w", err))
+			vuln.CvssV3 = cvssV3
+			// Overwrite CVSSV2 if already set.
+			vuln.ScoreVersion = sv
+			vuln.Cvss = score
 		}
 	}
-	if v3 := vulnCVSS.GetV3(); v3 != nil {
-		if c, err := cvssv3.ParseCVSSV3(v3.GetVector()); err == nil {
-			err = cvssv3.CalculateScores(c)
-			if err != nil {
-				errList.AddError(fmt.Errorf("calculating CVSS v3 scores: %w", err))
+
+	return errList.ToError()
+}
+
+func setScoresAndScoreVersionList(vuln *storage.EmbeddedVulnerability, cvssMetrics []*v4.VulnerabilityReport_Vulnerability_CVSS) error {
+	errList := errorhelpers.NewErrorList("failed to get CVSS Metrics")
+
+	if len(cvssMetrics) == 0 {
+		return nil
+	}
+
+	var scores []*storage.CVSSScore
+	for _, cvss := range cvssMetrics {
+		score := &storage.CVSSScore{
+			Source: storage.Source(cvss.Source.Number()),
+			Url:    cvss.Url,
+		}
+
+		if cvss.GetV3() != nil {
+			_, _, cvssV3, v3Err := toCVSSV3Scores(cvss, vuln.GetCve())
+			if v3Err == nil && cvssV3 != nil {
+				score.CvssScore = &storage.CVSSScore_Cvssv3{Cvssv3: cvssV3}
+				scores = append(scores, score)
+				continue
+			} else {
+				errList.AddError(v3Err)
 			}
-			// Use the report's score if it exists.
-			if baseScore := v3.GetBaseScore(); baseScore != 0.0 && baseScore != c.Score {
-				log.Debugf("Calculated CVSSv3 score does not match given base score (%f != %f) for %s. Using given score...", c.Score, baseScore, vuln.GetCve())
-				c.Score = baseScore
+		}
+
+		// Fallback to CVSSV2 if CVSSV3 is not available
+		if cvss.GetV2() != nil {
+			_, _, cvssV2, v2Err := toCVSSV2Scores(cvss, vuln.GetCve())
+			if v2Err == nil && cvssV2 != nil {
+				score.CvssScore = &storage.CVSSScore_Cvssv2{Cvssv2: cvssV2}
+				scores = append(scores, score)
+			} else {
+				errList.AddError(v2Err)
 			}
-			c.Severity = cvssv3.Severity(c.Score)
-			vuln.CvssV3 = c
-			// Overwrite v2, if set.
-			vuln.ScoreVersion = storage.EmbeddedVulnerability_V3
-			vuln.Cvss = c.Score
-		} else {
-			errList.AddError(fmt.Errorf("v3: %w", err))
 		}
 	}
+
+	if len(scores) > 0 {
+		vuln.CvssMetrics = scores
+		return nil
+	}
+
 	return errList.ToError()
 }
 
+func toCVSSV2Scores(vulnCVSS *v4.VulnerabilityReport_Vulnerability_CVSS, cve string) (float32, storage.EmbeddedVulnerability_ScoreVersion, *storage.CVSSV2, error) {
+	errList := errorhelpers.NewErrorList("failed to parse to CVSS V2 scores")
+	v2 := vulnCVSS.GetV2()
+	if c, err := cvssv2.ParseCVSSV2(v2.GetVector()); err == nil {
+		err = cvssv2.CalculateScores(c)
+		if err != nil {
+			errList.AddError(fmt.Errorf("calculating CVSS v2 scores: %w", err))
+			return 0, 0, nil, errList.ToError()
```

Comment:
nit: If returning right after error, you can use `errors.Wrapf(...)` in the return statement instead of first adding the error to a list and then returning `errList.ToError()`